### PR TITLE
gsl,tbb: split output

### PR DESCRIPTION
**Pull request by Artturin**
found with 
`nix path-info -r /run/current-system | xargs -i fd --type d --full-path 'include/' {}  | rg -v "\-dev" | xargs du -shc --apparent-size | sort -hk1`

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Artturin** [comment]
other large include dirs

`nix path-info -r /run/current-system | xargs -i fd --type d --full-path 'include/' {}  | rg -v "\-dev" | xargs du -shc --apparent-size | sort -hk1`

```
389K    /nix/store/3bvlq0akai4x89jsn2dlvlikk27r93zi-libplacebo-4.208.0/include/libplacebo/
393K    /nix/store/rv6badsn64d17accb2iwkxfqngx7k2as-libbpf-1.1.0/include/linux/
398K    /nix/store/6d9cdiv9dlg635wfazr1bh4x5x4sq5jx-rocm-llvm-llvm-5.4.2/include/llvm-c/
411K    /nix/store/znwrpnp8dlh3b9zpgpm1i7c4nb51z93f-libvirt-8.10.0/include/libvirt/
432K    /nix/store/awylv4dyczcwa4qy9fx06nc87pbb72kd-aws-c-common-0.8.9/include/aws/
446K    /nix/store/lpir61yxxjhymlxy74ycgxqfdr86yjys-qca-qt5-2.3.4/include/Qca-qt5/
469K    /nix/store/3bfvwjayikvaz17lba8794bjvxm58nz8-modemmanager-1.18.12/include/libmm-glib/
480K    /nix/store/k45cjwh35qx1bdl4clg6mq6i0kxl3p8a-rocm-runtime-5.4.1/include/hsa/
509K    /nix/store/31idnad0cg2jhv5qs6y7fq5aw8q9g7g0-libyuv-1787/include/libyuv/
547K    /nix/store/zzwqcifb2hgm1qii6ixh86cjryzlf5zh-linux-headers-6.1/include/drm/
631K    /nix/store/yyahnn0xnwcshm8a21zqgg5b4pca7z7f-rocm-llvm-llvm-5.4.2/include/sanitizer/
638K    /nix/store/g9qkr3ncgh27kwsynq03r75amd9i7fwj-sqlcipher-4.5.3/include/sqlcipher/
645K    /nix/store/hyfid0yaiqgdp0hjy0z5xf85v0dqrp7h-taglib-1.13/include/taglib/
725K    /nix/store/abax98471z8fshv4b9p46bkh3lxmpy0z-python3-3.10.9/include/python3.10/
773K    /nix/store/fvgl8js6f2ig5f2w3m746ya8jzza1ihz-directfb-1.7.7/include/directfb-internal/
841K    /nix/store/kg6qk2qrjvfhqyhkmgvi1h193fj48qc3-rocm-opencl-runtime-5.4.2/include/CL/
852K    /nix/store/fvgl8js6f2ig5f2w3m746ya8jzza1ihz-directfb-1.7.7/include/directfb/
887K    /nix/store/ykm2v353pwr9yyj4vpgqxbncn3r6d56a-nlohmann_json-3.11.2/include/nlohmann/
912K    /nix/store/b19kysh00skk7n8fn676q69sxn5v2qhq-python3.10-mypy-0.991/lib/python3.10/site-packages/mypyc/external/googletest/include/gtest/
1.1M    /nix/store/w90327cv9z2q6czjhzsfh7hs3fa8hs71-libsigc++-2.10.8/include/sigc++-2.0/
1.2M    /nix/store/fpyyvrmgpv9ggawnfa8i8zckmpqp79jq-xorgproto-2021.5/include/X11/
1.2M    /nix/store/nlc8fv3a5r2vr1vq07y5qw6riz7d6v8h-gsl-2.7.1/include/gsl/
1.8M    /nix/store/hcbalan0xq30hb53myvw93i309ifh4yg-tbb-2020.3/include/tbb/
2.5M    /nix/store/hr6kxafywg7ar3gpfjzplgzxcr31g23w-zfs-kernel-2.1.7-6.0.19/src/zfs-2.1.7/include/sys/
2.8M    /nix/store/7y1qagav6xv4k3w0df5d7z91y2d3w4z7-nodejs-18.13.0/include/node/
2.8M    /nix/store/g57a0hbcvx6ph0a144s0fq0n0j2pqww6-protobuf-3.21.12/include/google/
3.8M    /nix/store/zadsncz62r5nd6wfsw77754lva7hzqvq-opencv-3.4.18/include/opencv2/
4.5M    /nix/store/zzwqcifb2hgm1qii6ixh86cjryzlf5zh-linux-headers-6.1/include/linux/
5.2M    /nix/store/5wb6fwc2g1f1pjq5hasnlz7y64b8g7by-msgpack-3.3.0/include/msgpack/
6.1M    /nix/store/yyahnn0xnwcshm8a21zqgg5b4pca7z7f-rocm-llvm-llvm-5.4.2/include/c++/
6.8M    /nix/store/j38c32v53zmn3ipy9cc5g806fyxc5fvv-opencv-4.7.0/include/opencv4/
11M     /nix/store/z1yw9cw97bcdfmxz5wk7j1bycw42r1mw-gcc-11.3.0/include/c++/
18M     /nix/store/9c4r6fcn5h15l1hzywf7w8aaqyvd9fhi-rocm-llvm-clang-5.4.2/include/clang/
21M     /nix/store/6d9cdiv9dlg635wfazr1bh4x5x4sq5jx-rocm-llvm-llvm-5.4.2/include/llvm/
```

**Comment by markuskowa** [comment]
@GrahamcOfBorg build cp2k d-seams

**Comment by Artturin** [comment]
@markuskowa both build for me locally, are those just 2 packages you picked randomly or have they had issues when gsl/tbb have been updated

**Comment by markuskowa** [comment]
Yep, works for me too locally. 
Those are two packages I care about, that's why checked them. 
Good to merge from my side.

**Review by markuskowa — APPROVED** [review]

File: pkgs/development/libraries/gsl/default.nix [hunk 1]
```diff
@@ -9,10 +9,16 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-3LD71DBIgyt1f/mUJpGo3XACbV2g/4VgHlJof23us0s=";
   };
 
+  outputs = [ "out" "dev" ];
```
Comment (jtojnar) [inline pkgs/development/libraries/gsl/default.nix:12]: Most common convention is placing outputs above `src`, see https://github.com/jtojnar/nixpkgs-hammering/blob/0c54b7121a21f79e580773a46fc8d51482dc1d1c/explanations/attribute-ordering.md.

**Review by edwtjo — APPROVED** [review]
LGTM

**Pull request merged**